src/Routing/RouteBuilder.php

```diff
@@ -914,18 +912,12 @@ public function plugin(string $name, $options = [], $callback = null)
      * @return $this
      * @throws \InvalidArgumentException when there is no callable parameter.
      */
-    public function scope(string $path, $params, $callback = null)
+    public function scope(string $path, callable|array $params, ?callable $callback = null)
     {
-        if (!is_array($params)) {
+        if (is_callable($params)) {
             $callback = $params;
             $params = [];
         }
-        if (!is_callable($callback)) {
```

Comment:
Don't we still need this as $params could be an array, and $callback could be null.

Reply:
that's true.